### PR TITLE
SONARJAVA-1525 Handle properly relation between same symbolic values

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-squid/src/main/java/org/sonar/java/se/symbolicvalues/BinaryRelation.java [hunk 1]
```diff
@@ -93,6 +93,23 @@ protected RelationState resolveState(Collection<BinaryRelation> knownRelations)
 
   @CheckForNull
   protected RelationState resolveState(Collection<BinaryRelation> knownRelations, Set<BinaryRelation> usedRelations) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/se/symbolicvalues/BinaryRelation.java]: MAJOR The Cyclomatic Complexity of this method "resolveState" is 21 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-squid/src/test/files/se/ConditionAlwaysTrueOrFalseCheck.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
 @interface CheckForNull {}
 
 public static class Class extends SuperClass {
-
+/*
```
Comment (m-g-sonar) [inline java-squid/src/test/files/se/ConditionAlwaysTrueOrFalseCheck.java]: I'm not sure you wanted to comment everything here...

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**